Elevel/Application/Features/UserFeatures/GetCoachesQuery.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AutoMapper;
+using Elevel.Application.Extensions;
+using Elevel.Application.Interfaces;
+using Elevel.Domain.Enums;
+using Elevel.Domain.Models;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elevel.Application.Features.UserFeatures
+{
+    public class GetCoachesQuery
+    {
+        public class Request : IRequest<Response>
+        {
+            public string Name { get; set; }
+        }
+
+        public class Handler : IRequestHandler<Request, Response>
+        {
+            private readonly IMapper _mapper;
+            private readonly UserManager<User> _userManager;
+            private readonly IApplicationDbContext _context;
+
+            public Handler(IMapper mapper, UserManager<User> userManager, IApplicationDbContext context)
+            {
+                _mapper = mapper;
+                _userManager = userManager;
+                _context = context;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+                var coaches = await _userManager.Users.WhereAsync(async x => (await _userManager.GetRolesAsync(x)).Contains(UserRole.Coach.ToString()));
```

Comment:
ouch. why we can't truly setup identity tables ? there are tables with UserRole and Role tables and relations, so it can be im[plemented in 1 request with 2 includes instead of this approach with custom where and many requests to DB